### PR TITLE
Always queue project removal actions

### PR DESCRIPTION
**Pull request by mattwar**
This fixes VSO bug 296548: https://devdiv.visualstudio.com/DevDiv/_workitems?id=296548&_a=edit

The VSO bug is based on watson crash.  The culprit is a race condition in shutting down projects on close with other potential pending project affecting actions that are now being queued to run on UI thread idle.

@dotnet/roslyn-ide, @mavasani please review.

### CODE REVIEW

File: src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpReferencesTests.cs [hunk 1]
```diff
@@ -28,9 +28,6 @@ public void AddingReferenceToProjectMetadataPromotesToProjectReference()
                 project2.OnImportAdded(@"c:\project1.dll", "project1");
 
                 Assert.Equal(true, project2.GetCurrentProjectReferences().Any(pr => pr.ProjectId == project1.Id));
-
-                project2.Disconnect();
```
Comment (mavasani) [inline src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpReferencesTests.cs]: Any reason for removing the Disconnect calls?

Comment (mattwar) [inline src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpReferencesTests.cs]: Yes, the TestEnvironment was already calling Disconnect for all projects.  These lines caused Disconnect to be called multiple times on the same instance, which eventually lead to exceptions due to disposed objects being re-disposed.  When the project remove was immediate, this was circumvented somehow.

Comment (mavasani) [inline src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpReferencesTests.cs]: Thanks!

**Comment by mattwar** [comment]
@MattGertz approval?

**Comment by MattGertz** [comment]
Approved for RC.3

**Pull request merged**